tasks/licenses.js

```diff
@@ -10,75 +8,85 @@ export default function licenses(grunt) {
+      npmLicense.init(options, (result, error) => {
         cb(undefined, result);
       });
     });
 
-    // Only check production NPM dependencies, not dev
-    await fromNode(cb => npm.load({production: true}, cb));
-
     const npmList = await fromNode(cb => {
       npm.commands.list([], true, (a, b, npmList) => cb(undefined, npmList));
     });
 
-    // Recurse npm --production --json ls output, create array of dependency
-    // objects, each with a name prop formatted as 'package@version' and a
-    // path prop leading back to the root dependency.
-    function getDependencies(dependencies, path, list = []) {
-      Object.keys(dependencies).forEach(name => {
-        const dependency = dependencies[name];
-        const newPath = path ? `${path} -> ${dependency.from}` : dependency.from;
-        list.push({
-          path: newPath,
-          name: name + '@' + dependency.version
-        });
+    function flattenDependencyTree(deps, path = [], list = {}) {
+      Object.keys(deps).forEach(name => {
+        let dependency = deps[name];
+        let displayName = `${name}@${dependency.version}`;
+
+        list[displayName] = {
+          name: name,
+          version: dependency.version,
+          path: path
+        };
 
         if (dependency.dependencies) {
-          getDependencies(dependency.dependencies, newPath, list);
+          flattenDependencyTree(dependency.dependencies, path.concat(displayName), list);
         }
       });
+
       return list;
-    };
+    }
 
-    const productionDependencies = {};
-    getDependencies(npmList.dependencies).forEach(dependency => {
-      productionDependencies[dependency.name] =
-        Object.assign({}, allDependencies[dependency.name], dependency);
-    });
+    const allDependencies = flattenDependencyTree(npmList.dependencies);
+
+    /**
+     * Licenses for a package by name with overrides
+     *
+     * @param {String} name
+     * @return {Array}
+     */
+
+    function licensesForPackage(name) {
+      let licenses = packages[name].licenses;
+
+      if (config.overrides.hasOwnProperty(name)) {
+        licenses = config.overrides[name];
+      }
+
+      return typeof licenses === 'string' ? [licenses] : licenses;
+    }
+
+    /**
+     * Determine if a package has a valid license
+     *
+     * @param {String} name
+     * @return {Boolean}
+     */
+
+    function isInvalidLicense(name) {
+      let licenses = licensesForPackage(name);
+      return _.intersection(licenses, config.licenses).length !== licenses.length;
+    }
+
+    /**
+     * Generator for iterating over packages with invalid licenses
+     */
 
-    const licenseStats = _.map(productionDependencies, processPackage);
-    const invalidLicenses = licenseStats.filter(pkg => !pkg.valid);
+    let invalidLicenses = [];
 
-    if (!grunt.option('only-invalid')) {
-      grunt.log.debug(JSON.stringify(licenseStats, null, 2));
+    for (let name of Object.keys(packages)) {
```

Comment:
I ended up removing all of this by [removing the dependency tree](https://github.com/elastic/kibana/pull/7920/commits/43a94a5ffbb307a08aff0a511c194635a0c070f6). 

```
Running "licenses" task
kibana@5.0.0-alpha6 /Users/tyler/Code/tylersmalley/kibana
├─┬ babel-core@5.8.38
│ └── is-integer@1.0.6
├─┬ chokidar@1.6.0
│ └─┬ fsevents@1.0.14
│   └── tweetnacl@0.13.3
└─┬ webpack@1.12.1
  └─┬ node-libs-browser@0.6.0
    └─┬ http-browserify@1.7.0
      └── Base64@0.2.1

Warning: Non-confirming licenses:
 Base64@0.2.1
 is-integer@1.0.6
 tweetnacl@0.13.3

 Use --force to continue.

Aborted due to warnings.
```

Thoughts @epixa and @cjcenizal?